### ISSUE TITLE
NA values with stringsAsFactors = true

### ISSUE THREAD

**Issue by kungfujam** [Rdatatable/data.table#1466]
If I run `fread` specifying na.strings is this not respected by stringsAsFactors? I see in [the code](https://github.com/Rdatatable/data.table/blob/9b14ec740680a398d5db16f0f5716f02532b1095/R/fread.R#L117) that `as.factor` is called without the `exclude=` argument.

I'm asking because I get factors with `NA` as one of the levels. This means `DT[is.na(factorcol),]` returns no rows which is not what I was expecting. The workaround it to use `DT[is.na(levels(factorcol)),]`

**Comment by arunsrinivasan** [comment]
I don't fully understand this. There's no reproducible example or version number. But I think its duplicate of https://github.com/Rdatatable/data.table/pull/1408 and news item 14 under bug fixes for v1.9.7. We don't use `as.factor()` but our own `as_factor()`. 

Please try with devel version before posting.

```r
# v1.9.7 (latest commit)
require(data.table)
fread('a,b\na,2\nb,3\nc,4\nNA,5\n', stringsAsFactors=TRUE, na.strings=c("NA", "c"))[is.na(a)]
#     a b
# 1: NA 4
# 2: NA 5
```

If I seem to have misunderstood, please explain with a reproducible example. Thanks.

**Comment by kungfujam** [comment]
Exactly what I needed. Many thanks.
